cirq/contrib/noise_models/noise_models.py

```diff
@@ -0,0 +1,152 @@
+# Copyright 2019 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Sequence, TYPE_CHECKING
+
+from cirq import devices, value, ops, protocols
+
+if TYPE_CHECKING:
+    import cirq
+
+
+def _moment_is_measurements(moment: 'cirq.Moment') -> bool:
+    """Whether the moment is nothing but measurement gates.
+
+    If a moment is a mixture of measurement and non-measurement gates
+    this will throw a ValueError.
+    """
+    is_measurements = None
+    for gate in moment:
+        if protocols.is_measurement(gate):
+            if is_measurements is None:
+                is_measurements = True
+            elif is_measurements:
+                # Check that *all* are measurements
+                pass
+            else:
+                raise ValueError("Moment must be all measurements "
+                                 "or all operations.")
+        else:
+            if is_measurements is None:
+                is_measurements = False
+            elif not is_measurements:
+                # Check that all are *not* measurements
+                pass
+            else:
+                raise ValueError("Moment must be all measurements "
+                                 "or all operations.")
+    return is_measurements
+
+
+class DepolarizingNoiseModel(devices.NoiseModel):
+    """Applies depolarizing noise to each qubit individually at the end of
+    every moment.
+
+    If a circuit contains measurements, they be in moments that don't
```

Comment:
grammar

Reply:
I was trying to be hip.

Jk, good catch